internal/buf/bufimage/bufimagebuild/bufimagebuildtesting/bufimagebuildtesting.go

```diff
@@ -15,84 +15,224 @@
 package bufimagebuildtesting
 
 import (
-	"bytes"
 	"context"
 	"fmt"
-	"io"
+	"os"
+	"path/filepath"
 	"strings"
 
+	"github.com/bufbuild/buf/internal/buf/bufanalysis"
+	"github.com/bufbuild/buf/internal/buf/bufimage"
 	"github.com/bufbuild/buf/internal/buf/bufimage/bufimagebuild"
 	"github.com/bufbuild/buf/internal/buf/bufmodule"
 	"github.com/bufbuild/buf/internal/buf/bufmodule/bufmodulebuild"
-	"github.com/bufbuild/buf/internal/pkg/storage/storagemem"
+	"github.com/bufbuild/buf/internal/buf/internal/buftesting"
+	"github.com/bufbuild/buf/internal/pkg/prototesting"
+	"github.com/bufbuild/buf/internal/pkg/storage/storageos"
+	"github.com/bufbuild/buf/internal/pkg/tmp"
+	"go.uber.org/multierr"
 	"go.uber.org/zap"
+	"golang.org/x/tools/txtar"
+	"google.golang.org/protobuf/types/descriptorpb"
 )
 
-const fileSplitter = "----------------\n"
-
 // Fuzz is the entrypoint for the fuzzer.
 // We use https://github.com/dvyukov/go-fuzz for fuzzing.
 // Please follow the instructions
 // in their README for help with running the fuzz targets.
 func Fuzz(data []byte) int {
-	i, _ := fuzz(data)
-	return i
+	ctx := context.Background()
+	result, err := fuzz(ctx, data)
+	if err != nil {
+		// data was invalid in some way
+		return -1
+	}
+	return result.panicOrN(ctx)
 }
 
-func fuzz(data []byte) (int, error) {
-	files := bytes.Split(data, []byte(fileSplitter))
-	pathToData := make(map[string][]byte)
-	for _, file := range files {
-		buf := bytes.NewBuffer(file)
-		header, err := buf.ReadBytes('\n')
-		if err != nil {
-			return 0, err
-		}
-		content, err := io.ReadAll(buf)
-		if err != nil {
-			return 0, err
-		}
-		// Trim comment prefix and newline suffix
-		fileName := strings.TrimSpace(strings.TrimPrefix(string(header), "// "))
-		pathToData[fileName] = content
+func fuzz(ctx context.Context, data []byte) (_ *fuzzResult, retErr error) {
+	dir, err := tmp.NewDir()
+	if err != nil {
+		return nil, err
+	}
+	defer func() {
+		retErr = multierr.Append(retErr, dir.Close())
+	}()
+	if err := untxtar(data, dir.AbsPath()); err != nil {
+		return nil, err
 	}
-	bucket, err := storagemem.NewReadBucket(pathToData)
+
+	filePaths, err := buftesting.GetProtocFilePathsErr(ctx, dir.AbsPath(), 0)
 	if err != nil {
-		return 0, err
+		return nil, err
+	}
+
+	actualProtocFileDescriptorSet, protocErr := prototesting.GetProtocFileDescriptorSet(
+		ctx,
+		[]string{dir.AbsPath()},
+		filePaths,
+		false,
+		false,
+	)
+
+	image, bufAnnotations, bufErr := fuzzBuild(ctx, dir.AbsPath())
+	return newFuzzResult(
+		bufAnnotations,
+		bufErr,
+		protocErr,
+		actualProtocFileDescriptorSet,
+		image), nil
```

Comment:
Style nit: put the closing bracket on a new line 😁 
```suggestion
		image,
	), nil
```